diff --git a/src/crypto.js b/src/crypto.js
index 3896ada0c606beb9f9b77faa5831114a12a2214a..38b64463618b932c7d7139f91b58eda992269f5f 100644
--- a/src/crypto.js
+++ b/src/crypto.js
@@ -124,7 +124,8 @@ function performDetectPrivateKeyAlgorithm(key) {
 }
 
 function performDetectPublicKeyAlgorithms(key) {
-  const publicKeyPemMatch = key.match(publicKeyPemMatcher)
+  const trimmedKey = key.trim()
+  const publicKeyPemMatch = trimmedKey.match(publicKeyPemMatcher)
 
   if (key.match(privateKeyPemMatcher)) {
     throw new TokenError(TokenError.codes.invalidKey, 'Private keys are not supported for verifying.')
